src/main/java/org/example/Main.java

```diff
@@ -13,46 +20,46 @@ public static boolean validParentheses(String input) {
         //Trigger condition(false): the following bracket, if a closed bracket; must be the same as the last opening bracket
         //if condition not triggered; return true
         //check first bracket, if close bracket then return false
-
+        if (input == null || input.length() == 0) {
+            return false;
+        }
         Stack<Character> stack = new Stack<>();
-        char c = input.charAt(0);
-        if(c == ')' || c == ']' || c == '}'){ //if first char is closed, it's false
+        char first = input.charAt(0);
+        if(first == ')' || first == ']' || first == '}'){ //if first char is closed, it's false
             return false;
-        }else{
-            stack.push(c); //add c to stack
+        } else {
+            stack.push(first); //add c to stack
         }
 
-        for(int i=1; i<input.length(); i++){ //loop through each char in the String
+        for(int i = 1; i < input.length(); i++){ //loop through each char in the String
             //get the last open bracket stored
-            char c1 = input.charAt(i);
-            //check if c1 is a open bracket
-            if(c1!='(' && c1!='[' && c1!='{' && c1!=')' && c1!=']' && c1!='}'){ //check for non-brackets, skip if true
+            char current = input.charAt(i);
+            //check if current is a open bracket
+            boolean isOpenBracket = current =='(' || current =='[' || current =='{';
+            boolean isCloseBracket = current ==')' || current ==']' || current =='}';
+            if(!isOpenBracket && !isCloseBracket){ //check for non-brackets, skip if true
```

Comment:
Long if statement is usually hard to read. Use variable names to communicate intent. Ideally, this should be a private function.

For example, if you want use a if statement of check if integer is odd number, you should write 
```
Boolean isOdd = i % 2 == 1;
if (isOdd) {
  // some code
}
```
or 

```
if (isOdd(i) {
  // some code
}

private isOdd(int i) {
  // some private function
}
```